data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/beacon/SwaggerIntegrationTest.java

```diff
@@ -0,0 +1,114 @@
+  private JsonNode currentJsonNodes;
+
+  @BeforeEach
+  public void setup() throws IOException {
+    startRestAPIAtGenesis();
+    final Response response = get();
+    currentJsonNodes = OBJECT_MAPPER.readTree(response.body().string());
+  }
+
+  @Test
+  void schemaObjects_shouldBeConsistent(@TempDir final Path tempDir) throws IOException {
+    final JsonNode schemas = currentJsonNodes.findPath("components").findPath("schemas");
+
+    List<String> fieldNames = ImmutableList.copyOf(schemas.fieldNames());
+    for (String current : fieldNames) {
+      checkStructureIsConsistent("schema", tempDir, current, schemas.path(current));
+    }
+    checkAllExpectedFilesExist(tempDir, "schema");
+  }
+
+  @Test
+  void paths_shouldBeConsistent(@TempDir final Path tempDir) throws IOException {
+    final JsonNode paths = currentJsonNodes.findPath("paths");
+
+    List<String> fieldNames = ImmutableList.copyOf(paths.fieldNames());
+    for (String current : fieldNames) {
+      checkStructureIsConsistent("paths", tempDir, current, paths.path(current));
+    }
+
+    checkAllExpectedFilesExist(tempDir, "paths");
+  }
+
+  private void checkAllExpectedFilesExist(final Path tempDir, final String schema)
+      throws IOException {
+    final List<String> expectedFileListing =
+        Arrays.asList(
+            Resources.toString(
+                    Resources.getResource(SwaggerIntegrationTest.class, schema + "_listing.txt"),
+                    UTF_8)
+                .split("\n"));
+    for (String file : expectedFileListing) {
+      assertThat(tempDir.resolve(file).toFile()).exists();
+    }
+  }
+
+  private void checkStructureIsConsistent(
+      final String namespace, final Path tempDir, final String elementName, final JsonNode node)
+      throws IOException {
+    final String filename = elementName.replaceAll("/", "_") + ".json";
+    Files.write(tempDir.resolve(filename), prettyJson(node).getBytes(UTF_8));
+    final JsonNode expectedNode = loadResourceFile(namespace + "/" + filename);
+    assertThat(node)
+        .withFailMessage(
+            String.format(
+                "Expected structure of %s: %s to match that stored in %s:%s",
+                elementName,
+                prettyJson(node),
+                namespace + "/" + filename,
+                prettyJson(expectedNode)))
+        .isEqualTo(expectedNode);
+  }
+
+  private String prettyJson(final JsonNode jsonNode) throws JsonProcessingException {
+    System.out.println(OBJECT_MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(jsonNode));
+    return OBJECT_MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(jsonNode);
+  }
+
+  private JsonNode loadResourceFile(final String resourceFile) throws IOException {
+    final String resource =
+        Resources.toString(
+            Resources.getResource(SwaggerIntegrationTest.class, resourceFile), UTF_8);
+    return OBJECT_MAPPER.readTree(resource);
+  }
+
+  public Response get() throws IOException {
```

Comment:
nit: Maybe `getOpenApi()` or something like that?

Reply:
it's just how all the integration tests are written, but I'm happy to rename this one

Reply:
I'm not overly fussed if that's what we do.